src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManagerProvider.cs

```diff
@@ -23,6 +25,21 @@ internal class ItemManagerProvider : IAsyncCompletionItemManagerProvider
         public ItemManagerProvider(RecentItemsManager recentItemsManager)
             => _instance = new ItemManager(recentItemsManager);
 
-        public IAsyncCompletionItemManager GetOrCreate(ITextView textView) => _instance;
+        public IAsyncCompletionItemManager GetOrCreate(ITextView textView)
+        {
+            if (textView.TextBuffer.TryGetWorkspace(out var workspace))
+            {
+                var workspaceContextService = workspace.Services.GetRequiredService<IWorkspaceContextService>();
+
+                // If we're in a cloud environment context, we want to avoid returning a completion item manager.
+                // Otherwise, we'll interfere with the LSP client manager and disrupt filtering.
```

Comment:
i don't really understand the comment.  why does this happen?  what is going on here?  this seems sooo hacky.

Reply:
@CyrusNajmabadi 
The lsp client provides their own implementation of the editor async completion APIs that then call the LSP servers for the completion results.  The bug here is that Roslyn's async completion is also active on the client (but doesn't actually provide any completion results).  When the LSP client pops up the completion list with results and the user starts typing to filter, Roslyn's implementation for sorting and filtering takes over when it should be the LSP client handling it.  Is that summary basically correct @AmadeusW ?

The change here disables our async completion when in a connected client context.  Happy to take other suggestions on how to fix this though!

Reply:
@dibarbet Isn't this why we have a separate content type though? I get we're trying to delete that, but why is this bug happening now?

Reply:
@jasonmalinowski this is exported for the Roslyn content type, which C#_LSP inherits from

Reply:
Ah, right boo. :-(